services/docker/docker-size.service.js

```diff
@@ -20,14 +27,40 @@ const pagedSchema = Joi.object({
     Joi.object({
       name: Joi.string().required(),
       full_size: nonNegativeInteger.required(),
+      images: Joi.array().items(
+        Joi.object({
+          size: nonNegativeInteger.required(),
+          architecture: Joi.string().required(),
+        })
+      ),
     })
   ),
 }).required()
 
 const queryParamSchema = Joi.object({
   sort: Joi.string().valid('date', 'semver').default('date'),
+  arch: Joi.alternatives([
+    Joi.string().valid(...validArchitectures),
+    Joi.number().valid(...validArchitectures),
+  ]),
 }).required()
 
+// If user provided the arch parameter,
+// check if any of the returned images has an architecture matching the arch parameter provided.
+// If yes, return the size of the image with this arch.
+// If not, return the value of the `full_size` in the latestEntry from the response.
+// For details see: https://github.com/badges/shields/issues/8238
+function getImageForArch(images, arch) {
+  const imgWithArch = Object.values(images).find(
+    img => img.architecture === arch
+  )
+
+  if (!imgWithArch) {
+    throw new NotFound({ prettyMessage: 'architecture not found' })
+  }
+  return imgWithArch.size
```

Comment:
The thing you're actually getting/returning here is a size, not an image
This should either be called `getImageForArch` and return an image object
Or it should be called `getImageSizeForArch` and return a size

Reply:
Good point, updated, thanks